quiz.js

```diff
@@ -0,0 +1,47 @@
+
+// JAVASCRIPT FOR MAKING OUR TREE GROW
+
+// We want spaces to equal the tree height - 1.
+// We want the tree characters to equal double the height - 1.
+  // OR we want it to equal loopCount + 1
+// Then we need to concatinate the whole thing to equal one line of code on the console log.
+
+// EVENT LISTENER (ONLY APPLIES TO SUBMIT BUTTON)
+var createTree = document.getElementById("button");
+createTree.addEventListener("click", makeTree);
+
+// FUNCTION FOR MAKING TREE GROW
+function makeTree( event ) {
+  // Only necessary because I put everything in a form
+  event.preventDefault();
+  // This app would have still worked without placing the form values in an object,
+  // and leaving them as plain variables, but still need the practice for the future.
+  var tree = {};
+  tree.height = document.getElementById("height").value;
+  tree.character = document.getElementById("character").value;
```

Comment:
On lines 20 & 21, you grab the two input elements from the page, and immediately grab their values.  Then on lines 31, 34, 37, 42 & 43 you grab the exact same elements again (and again).  Each of these reads from the DOM is (relatively) slow.

Best practice is to just grab the element once and stick it in a variable: ```heightBox = document.getElementById("height")``` and then, every time you need to do something with it afterwards, you can address the individual property you want to get/set by just hitting up that variable: ```tree.height = heightBox.value```, thus only hitting the DOM once, but still having access to that element you grabbed further down your script.